### PR TITLE
Add icons and display appropriate fields for new event types

### PR DESCRIPTION
**Pull request by kaja-osojnik**
Related: https://github.com/oasisprotocol/explorer/pull/893 https://github.com/oasisprotocol/explorer/pull/906

Screenshots:

Before:
Screenshot 2023-12-08 at 17 10 38
Screenshot 2023-12-08 at 17 11 01

After:
Screenshot 2023-12-08 at 17 11 40
Screenshot 2023-12-08 at 17 11 57

### CODE REVIEW

**Comment by csillag** [comment]
What about [CustomIcons/Undelegate.tsx](https://github.com/oasisprotocol/explorer/tree/dcdab227fffce0ae9e1e42cbe9e758b9d67be62d/src/app/components/CustomIcons/Undelegate.tsx) added earlier? Isn't it obsolete now that we have the proper `_start` and _`done` events?

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx [hunk 1]
```diff
@@ -37,9 +39,9 @@ export const EventTypeIcon: FC<{
     [RuntimeEventType.coregas_used]: <></>,
     [RuntimeEventType.consensus_accountswithdraw]: <WithdrawIcon fontSize="inherit" />,
     [RuntimeEventType.consensus_accountsdeposit]: <DepositIcon fontSize="inherit" />,
-    [RuntimeEventType.consensus_accountsdelegate]: <UnknownIcon fontSize="inherit" />, // TODO: use correct icon
-    [RuntimeEventType.consensus_accountsundelegate_start]: <UnknownIcon fontSize="inherit" />, // TODO: use correct icon
-    [RuntimeEventType.consensus_accountsundelegate_done]: <UnknownIcon fontSize="inherit" />, // TODO: use correct icon
+    [RuntimeEventType.consensus_accountsdelegate]: <DelegateIcon fontSize="inherit" />, // TODO: use correct icon
+    [RuntimeEventType.consensus_accountsundelegate_start]: <UndelegateStartIcon fontSize="inherit" />,
+    [RuntimeEventType.consensus_accountsundelegate_done]: <UndelegateFinishIcon fontSize="inherit" />, // TODO: use correct icon
```
Comment (csillag) [inline src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:44]: The TODO note should be removed here

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx [hunk 2]
```diff
@@ -37,9 +39,9 @@ export const EventTypeIcon: FC<{
     [RuntimeEventType.coregas_used]: <></>,
     [RuntimeEventType.consensus_accountswithdraw]: <WithdrawIcon fontSize="inherit" />,
     [RuntimeEventType.consensus_accountsdeposit]: <DepositIcon fontSize="inherit" />,
-    [RuntimeEventType.consensus_accountsdelegate]: <UnknownIcon fontSize="inherit" />, // TODO: use correct icon
-    [RuntimeEventType.consensus_accountsundelegate_start]: <UnknownIcon fontSize="inherit" />, // TODO: use correct icon
-    [RuntimeEventType.consensus_accountsundelegate_done]: <UnknownIcon fontSize="inherit" />, // TODO: use correct icon
+    [RuntimeEventType.consensus_accountsdelegate]: <DelegateIcon fontSize="inherit" />, // TODO: use correct icon
```
Comment (csillag) [inline src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:42]: The TODO note should be removed here

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx [hunk 3]
```diff
@@ -266,6 +266,46 @@ export const RuntimeEventDetails: FC<{
...
+                    ...getPreciseNumberFormat(event.body.amount.Amount),
+                    ticker: event.body.amount?.Denomination,
+                  })}
+                </dd>
+              </>
+            )}
+          </StyledDescriptionList>
+        </div>
+      )
+    case RuntimeEventType.consensus_accountsdelegate: // TODO show this properly
```
Comment (csillag) [inline src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:282]: The TODO note should be removed here

File: src/oasis-nexus/api.ts [hunk 4]
```diff
@@ -556,11 +556,13 @@ export const useGetRuntimeEvents: typeof generated.useGetRuntimeEvents = (
...
-                event.type === 'accounts.burn' ||
-                event.type === 'consensus_accounts.deposit' ||
-                event.type === 'consensus_accounts.withdraw'
+                event.type === RuntimeEventType.accountstransfer ||
+                event.type === RuntimeEventType.accountsmint ||
+                event.type === RuntimeEventType.accountsburn ||
+                event.type === RuntimeEventType.consensus_accountsdeposit ||
+                event.type === RuntimeEventType.consensus_accountswithdraw ||
+                event.type === RuntimeEventType.consensus_accountsdelegate ||
+                event.type === RuntimeEventType.consensus_accountsundelegate_done
```
Comment (csillag) [inline src/oasis-nexus/api.ts:565]: Don't we also need `accountsundelete_start` here? (I don't know, just asking')

Comment (kaja-osojnik) [reply src/oasis-nexus/api.ts:565]: These are just the ones that have 'Amount' in the event type, which `accountsundelete_start`  doesn't.

**Review by lubej — APPROVED** [review]
LGTM, needs rebase

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx [hunk 5]
```diff
@@ -279,6 +319,88 @@ export const RuntimeEventDetails: FC<{
...
+            <dt>{t('common.to')}</dt>
+            <dd>
+              <AccountLink
+                address={event.body.to}
+                scope={scope}
+                plain={addressSwitchOption !== AddressSwitchOption.Oasis}
+              />
+              {addressSwitchOption === AddressSwitchOption.Oasis && <CopyToClipboard value={event.body.to} />}
+            </dd>
+            {/* TODO check is needed because some consensus events temporarily use this for rendering */}
```
Comment (lubej) [inline src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:347]: Not sure what this is about.

File: src/locales/en/translation.json [hunk 6]
```diff
@@ -323,7 +323,9 @@
     "gasUsed": "Gas used",
     "fields": {
       "amount": "Amount",
-      "owner": "Owner"
+      "owner": "Owner",
+      "activeShares": "Active Shares",
+      "deboundingShares": "Debounding Shares"
```
Comment (lukaw3d) [inline src/locales/en/translation.json:328]:
```suggestion
      "debondingShares": "Debonding Shares"
```

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx (hunk 5 above)
Comment (lukaw3d) [reply src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:347]: Leftover from https://github.com/oasisprotocol/explorer/pull/1012/files#diff-63da09cea2b3cda0ceb01e5a7cb74bfd43582ae8b89fcca1e63d5c0adca46514R267. It should be reversed and removed now

File: src/app/components/RuntimeTransactionLabel/index.tsx [hunk 7]
```diff
@@ -10,7 +10,7 @@ import { DepositIcon } from './../CustomIcons/Deposit'
 import { WithdrawIcon } from './../CustomIcons/Withdraw'
 import { TransferIcon } from './../CustomIcons/Transfer'
 import { DelegateIcon } from '../CustomIcons/Delegate'
-import { UndelegateIcon } from '../CustomIcons/Undelegate'
+import { UndelegateStartIcon } from '../CustomIcons/UndelegateStart'
```
Comment (lukaw3d) [inline src/app/components/RuntimeTransactionLabel/index.tsx:13]: Can Undelegate icon be deleted now?

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx (hunk 5 above)
Comment (lukaw3d) [inline src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:347]: Not needed

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx [hunk 8]
```diff
@@ -279,6 +319,88 @@ export const RuntimeEventDetails: FC<{
...
+            {event.body.amount?.Amount && event.body.amount.Denomination && (
+              <>
```
Comment (lukaw3d) [inline src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:385]: Not needed

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx [hunk 9]
```diff
@@ -266,6 +266,46 @@ export const RuntimeEventDetails: FC<{
             </dd>
             {/* TODO check is needed because some consensus events temporarily use this for rendering */}
             {event.body.amount?.Amount && event.body.amount?.Denomination && (
+              <>
```
Comment (lukaw3d) [inline src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:269]: Not needed anymore

File: src/app/components/RuntimeEvents/RuntimeEventDetails.tsx [hunk 10]
```diff
@@ -266,6 +266,46 @@ export const RuntimeEventDetails: FC<{
...
+              <AccountLink
+                address={event.body.to}
+                scope={scope}
+                plain={addressSwitchOption !== AddressSwitchOption.Oasis}
+              />
+              {addressSwitchOption === AddressSwitchOption.Oasis && <CopyToClipboard value={event.body.to} />}
+            </dd>
+
+            {event.body.amount?.Amount && event.body.amount.Denomination && (
               <>
```
Comment (lukaw3d) [inline src/app/components/RuntimeEvents/RuntimeEventDetails.tsx:309]: Not needed

**Pull request merged**